dashboard/test/models/script_test.rb

```diff
@@ -2061,25 +2061,25 @@ class SummarizeVisibleAfterScriptTests < ActiveSupport::TestCase
   end
 
   test "self.valid_scripts: returns alternate unit if user has a course experiment with an alternate unit" do
+    Script.destroy_all
     user = create(:user)
-    unit = create(:script)
-    alternate_unit = build(:script)
+    create(:script, published_state: SharedConstants::PUBLISHED_STATE.stable, name: 'original-unit')
+    alternate_unit = build(:script, published_state: SharedConstants::PUBLISHED_STATE.stable, name: 'alternate-unit')
 
     UnitGroup.stubs(:has_any_course_experiments?).returns(true)
-    Rails.cache.stubs(:fetch).returns([unit])
-    unit.stubs(:alternate_script).returns(alternate_unit)
+    Script.any_instance.stubs(:alternate_script).returns(alternate_unit)
 
     units = Script.valid_scripts(user)
     assert_equal [alternate_unit], units
   end
 
   test "self.valid_scripts: returns original unit if user has a course experiment with no alternate unit" do
+    Script.destroy_all
     user = create(:user)
-    unit = create(:script)
+    unit = create(:script, published_state: SharedConstants::PUBLISHED_STATE.stable)
 
     UnitGroup.stubs(:has_any_course_experiments?).returns(true)
-    Rails.cache.stubs(:fetch).returns([unit])
-    unit.stubs(:alternate_script).returns(nil)
+    Script.any_instance.stubs(:alternate_script).returns(nil)
```

Comment:
these two test cases are a mess, because they stub things on the `Script` class, which is the class being tested. I don't want to invest more time cleaning this up because we want to get rid of alternate units as soon as we can.